CODINGSTYLE.md

```diff
@@ -0,0 +1,544 @@
+val f = { (x: Int) =>
+  body
+}
+```
+
+If the first line ends up being two long, the parameter list should go the next line, and the body indented with two more spaces:
+
+```scala
+val someLongIdentifierWithHighIdentation = {
+  (x: Int, ys: List[Traversable[String]]) =>
+    body
+}
+```
+
+### Spaces
+
+There must not be any space before the following tokens: `: , ; )`
+
+There must be exactly one space after the following tokens: `: , ; if for while`
+Sometimes, it is acceptable to have several spaces, for vertical alignment reasons.
+
+There must be exactly one space before and after `{` and `}`.
+With the exception of partial import, where there is no space on either side.
+
+Binary operators, including `=>`, must have a single space on both sides.
+Sometimes, spaces can be removed to highlight the relatively higher priority wrt. to a neighboring operator, for easier visual parsing.
+For example, instead of `x < len - 1`, it is better to write `x < len-1`, highlighting that `-` has a higher priority than `<`.
+
+Unary operators must not be followed by a space.
+
+### Method call style
+
+Usually, parentheses should be used for actual parameters to a method call.
+Braces should be used instead if an argument list has only a lambda, and that lambda does not fit in an inline one-liner.
+
+In general, dot-notation should be used for non-symbolic methods, and infix notation should be used for symbolic methods.
+Infix notation is also used if the only argument is a brace lambda.
+
+Examples:
+
+```scala
+// inline lambda, hence (), hence dot-notation
+list.map(x => x * 2)
+
+// long lambda, hence braces, hence infix notation
+list map { x =>
+  if (x < 5) x
+  else x * 2
+}
+
+// symbolic operator, hence infix notation
+value :: list
+```
+
+Using dot-notation with a brace lambda is possible to force priorities.
+This is typically the case if the call is chained to a parameterless method call, as in
+
+```scala
+list.map { x =>
+  // complicated stuff
+}.toMap
+```
+
+When calling a method declared with an empty pair of parentheses, use `()`, except if the method is Java-defined *and* does not have side-effects.
+
+### Method definition
+
+All public and protected methods must have an explicit result type.
+
+Procedure syntax must not be used.
+`: Unit =` must be used instead.
+
+Side-effect-free methods without formal parameters should be declared without `()`, unless either a) it overrides a method defined with `()` (such as `toString()`) or b) it implements a Java method in the Java libraries.
+
+The signature of a method is technically a single line, and hence, if it has to be broken due to line length reasons, subsequent lines should be indented 4 spaces.
+As a reminder, the line can be broken right after a `,` or a `(` (and not, for example, after `implicit` or `:`).
+You should avoid breaking the line between the last parameter and the result type; going over the 80 characters limit is preferred in that case.
+
+### for comprehensions
```

Comment:
There is a slight ambiguity between the word "for" and the keyword "for". You should put `for` instead.
```
### `for` comprehensions
```